src/sxg_sig.c

```diff
@@ -22,71 +22,116 @@
 
 #include "libsxg/internal/sxg_buffer.h"
 #include "libsxg/internal/sxg_codec.h"
+#include "libsxg/sxg_buffer.h"
 
 sxg_sig_t sxg_empty_sig() {
   static const sxg_sig_t sig = {
-      .name = {.data = NULL, .size = 0, .capacity = 0},
-      .cert_sha256 = {.data = NULL, .size = 0, .capacity = 0},
-      .cert_url = {.data = NULL, .size = 0, .capacity = 0},
-      .ed25519key = {.data = NULL, .size = 0, .capacity = 0},
+      .name = NULL,
+      .name_size = 0,
+      .cert_sha256 = NULL,
+      .cert_sha256_size = 0,
+      .cert_url = NULL,
+      .cert_url_size = 0,
+      .ed25519key = NULL,
+      .ed25519key_size = 0,
       .date = 0,
       .expires = 0,
-      .integrity = {.data = NULL, .size = 0, .capacity = 0},
-      .sig = {.data = NULL, .size = 0, .capacity = 0},
-      .validity_url = {.data = NULL, .size = 0, .capacity = 0}};
+      .integrity = NULL,
+      .integrity_size = 0,
+      .sig = NULL,
+      .sig_size = 0,
+  };
   return sig;
 }
 
 void sxg_sig_release(sxg_sig_t* target) {
-  sxg_buffer_release(&target->name);
-  sxg_buffer_release(&target->cert_sha256);
-  sxg_buffer_release(&target->cert_url);
-  sxg_buffer_release(&target->ed25519key);
+  OPENSSL_free(target->name);
+  target->name = NULL;
+  OPENSSL_free(target->cert_sha256);
+  target->cert_sha256 = NULL;
+  OPENSSL_free(target->cert_url);
+  target->cert_url = NULL;
+  OPENSSL_free(target->ed25519key);
+  target->ed25519key = NULL;
   target->date = 0;
   target->expires = 0;
-  sxg_buffer_release(&target->integrity);
-  sxg_buffer_release(&target->sig);
-  sxg_buffer_release(&target->validity_url);
+  OPENSSL_free(target->integrity);
+  target->integrity = NULL;
+  OPENSSL_free(target->sig);
+  target->sig = NULL;
+  OPENSSL_free(target->validity_url);
+  target->validity_url = NULL;
 }
 
 bool sxg_sig_set_name(const char* name, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->name);
-  return sxg_write_string(name, &sig->name);
+  sig->name_size = strlen(name);
+  sig->name = OPENSSL_realloc(sig->name, sig->name_size + 1);
+  if (sig->name == NULL) {
+    return false;
+  }
+  strcpy(sig->name, name);
+  return true;
 }
 
 bool sxg_sig_set_cert_sha256(X509* certificate, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->cert_sha256);
-  return sxg_calculate_cert_sha256(certificate, &sig->cert_sha256);
+  sig->cert_sha256_size = SHA256_DIGEST_LENGTH;
```

Comment:
Isn't this always the same size?

Reply:
Yes, as long as SHA256, it must be the same size. But for consistency, I prefer to have it as a variable member rather than hard code into the logic.